core/src/main/scala/kafka/coordinator/group/GroupCoordinator.scala

```diff
@@ -375,9 +375,12 @@ class GroupCoordinator(val brokerId: Int,
       }
 
       if (eligibleGroups.nonEmpty) {
-        groupManager.cleanupGroupMetadata(None, eligibleGroups, Long.MaxValue)
+        // note: there is a lock on the group when the function 'removeAllOffsets()' is called inside 'cleanupGroupMetadata(...)'
```

Comment:
nit: I think mentioning this in the doc for `cleanupGroupMetadata` would be fine.